ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendProvider.scala

```diff
@@ -5,43 +5,96 @@ package com.daml.platform.store.backend
 
 import java.sql.Connection
 
+import com.daml.platform.store.backend.ParameterStorageBackend.LedgerEnd
 import com.daml.platform.store.backend.h2.H2StorageBackendFactory
 import com.daml.platform.store.backend.oracle.OracleStorageBackendFactory
 import com.daml.platform.store.backend.postgresql.PostgresStorageBackendFactory
+import com.daml.platform.store.cache.MutableLedgerEndCache
 import com.daml.testing.oracle.OracleAroundAll
 import com.daml.testing.postgresql.PostgresAroundAll
 import org.scalatest.Suite
 
-/** Creates a database and a [[StorageBackend]].
+/** Creates a database and a [[TestBackend]].
   * Used by [[StorageBackendSpec]] to run all StorageBackend tests on different databases.
   */
 private[backend] trait StorageBackendProvider {
   protected def jdbcUrl: String
-  protected def backendFactory: StorageBackendFactory
+  protected def backend: TestBackend
 
   protected final def ingest(dbDtos: Vector[DbDto], connection: Connection): Unit = {
     def typeBoundIngest[T](ingestionStorageBackend: IngestionStorageBackend[T]): Unit =
       ingestionStorageBackend.insertBatch(connection, ingestionStorageBackend.batch(dbDtos))
-    typeBoundIngest(backendFactory.createIngestionStorageBackend)
+    typeBoundIngest(backend.ingestion)
+  }
+
+  protected final def updateLedgerEnd(ledgerEnd: LedgerEnd)(connection: Connection): Unit = {
+    backend.parameter.updateLedgerEnd(ledgerEnd)(connection)
+    updateLedgerEndCache(connection)
```

Comment:
Out of interest. Is ledger end here and `head index` from `com.daml.platform.akkastreams.dispatcher.Dispatcher#getHead` related?

Reply:
yes they are, they are also very close (if not the same), also CacheIndex in mutable-state-cache is similar. we already have a task to understand/refactor/simplify if possible in this area.
as a first iteration we add this as an independent extension